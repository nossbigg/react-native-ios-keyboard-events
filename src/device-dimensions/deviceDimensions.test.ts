import { Dimensions, ScaledSize } from "react-native";
import {
  getDeviceModel,
  getDeviceOrientation,
  IDeviceModel,
} from "./deviceDimensions";

describe("#deviceDimensions", () => {
  const mockDimensionsGet = (height: number, width: number) => {
    jest
      .spyOn(Dimensions, "get")
      .mockReturnValue({ height, width } as ScaledSize);
  };

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("returns device orientation as landscape", () => {
    mockDimensionsGet(768, 1024);

    expect(getDeviceOrientation()).toBe("landscape");
    expect(Dimensions.get).toHaveBeenCalledWith("window");
  });

  it("returns device orientation as portrait", () => {
    mockDimensionsGet(1024, 768);
    expect(getDeviceOrientation()).toBe("portrait");
  });

  it("gets device model", () => {
    mockDimensionsGet(1024, 768);

    const deviceModel = getDeviceModel() as IDeviceModel;
    expect(deviceModel.model).toEqual("iPad Air");
    expect(Dimensions.get).toHaveBeenCalledWith("window");
  });

  it("does not get device model", () => {
    mockDimensionsGet(1, 1);
    expect(getDeviceModel()).toBe(undefined);
  });
});
