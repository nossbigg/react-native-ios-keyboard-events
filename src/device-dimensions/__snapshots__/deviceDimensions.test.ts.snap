// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#deviceDimensions contains device information of phone devices 1`] = `
Array [
  Object {
    "isTablet": false,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 199,
      },
      "portrait": Object {
        "docked": 253,
      },
    },
    "model": "iPhone 5s",
    "screenDimensions": Set {
      320,
      568,
    },
  },
  Object {
    "isTablet": false,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 200,
      },
      "portrait": Object {
        "docked": 260,
      },
    },
    "model": "iPhone 8",
    "screenDimensions": Set {
      375,
      667,
    },
  },
  Object {
    "isTablet": false,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 200,
      },
      "portrait": Object {
        "docked": 271,
      },
    },
    "model": "iPhone 8 Plus",
    "screenDimensions": Set {
      414,
      736,
    },
  },
  Object {
    "isTablet": false,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 209,
      },
      "portrait": Object {
        "docked": 335,
      },
    },
    "model": "iPhone X",
    "screenDimensions": Set {
      375,
      812,
    },
  },
  Object {
    "isTablet": false,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 209,
      },
      "portrait": Object {
        "docked": 346,
      },
    },
    "model": "iPhone XS Max",
    "screenDimensions": Set {
      414,
      896,
    },
  },
]
`;

exports[`#deviceDimensions contains device information of tablet devices 1`] = `
Array [
  Object {
    "isTablet": true,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 398,
        "minimized": 55,
        "split": 271,
      },
      "portrait": Object {
        "docked": 313,
        "minimized": 55,
        "split": 271,
      },
    },
    "model": "iPad Air",
    "screenDimensions": Set {
      1024,
      768,
    },
  },
  Object {
    "isTablet": true,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 398,
        "minimized": 55,
        "split": 271,
      },
      "portrait": Object {
        "docked": 313,
        "minimized": 55,
        "split": 271,
      },
    },
    "model": "iPad Pro (10.5-inch)",
    "screenDimensions": Set {
      1112,
      834,
    },
  },
  Object {
    "isTablet": true,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 418,
        "minimized": 69,
      },
      "portrait": Object {
        "docked": 333,
        "minimized": 69,
      },
    },
    "model": "iPad Pro (11-inch)",
    "screenDimensions": Set {
      1194,
      834,
    },
  },
  Object {
    "isTablet": true,
    "keyboardDimensions": Object {
      "landscape": Object {
        "docked": 471,
        "minimized": 55,
      },
      "portrait": Object {
        "docked": 378,
        "minimized": 55,
      },
    },
    "model": "iPad Pro (12.9-inch)",
    "screenDimensions": Set {
      1366,
      1024,
    },
  },
]
`;
